### PR TITLE
Add dependency information for initial RFR mail of a dependent PR

### PR DESCRIPTION
**Pull request by rwestberg**
When sending out an RFR email for a dependent PR, include a line describing the dependency.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Reviewers
 * [Erik Helin](https://openjdk.java.net/census#ehelin) (@edvbld - **Reviewer**) ⚠️ Review applies to 4e92b67c969ce22d3f7b32a36595440de2e7bacb

### Download
`$ git fetch https://git.openjdk.java.net/skara pull/991/head:pull/991`
`$ git checkout pull/991`

### CODE REVIEW

File: bots/mlbridge/src/main/java/org/openjdk/skara/bots/mlbridge/ArchiveMessages.java [hunk 1]
```diff
@@ -254,6 +254,16 @@ static String composeReplySubject(String parentSubject) {
...
     }
 
+    private static Optional<String> composeDependsOn(PullRequest pr) {
+        var dependsId = PreIntegrations.dependentPullRequestId(pr);
+        if (dependsId.isEmpty()) {
+            return Optional.empty();
+        }
+
+        var dependsPr = pr.repository().pullRequest(dependsId.get());
+        return Optional.of("Depends on PR: " + dependsPr.webUrl());
```
Comment (edvbld) [inline bots/mlbridge/src/main/java/org/openjdk/skara/bots/mlbridge/ArchiveMessages.java:264]:
```suggestion
        return Optional.of("Depends on: " + dependsPr.webUrl());
```

File: bots/mlbridge/src/test/java/org/openjdk/skara/bots/mlbridge/MailingListBridgeBotTests.java [hunk 2]
```diff
@@ -3195,4 +3205,76 @@ void rebaseOnRetry(TestInfo testInfo) throws IOException {
...
+            TestBotRunner.runPeriodicItems(mlBot);
+            listServer.processIncoming();
+
+            // Check the archive
+            Repository.materialize(archiveFolder.path(), archive.url(), "master");
+            assertTrue(archiveContains(archiveFolder.path(), "Subject: RFR: "), pr.id());
+            assertTrue(archiveContains(archiveFolder.path(), "Subject: Re: RFR: ", pr.id()));
+
+            assertTrue(archiveContains(archiveFolder.path(), "Depends on PR:", pr.id()));
+            assertFalse(archiveContains(archiveFolder.path(), "Depends on PR:", depPr.id()));
```
Comment (edvbld) [inline bots/mlbridge/src/test/java/org/openjdk/skara/bots/mlbridge/MailingListBridgeBotTests.java:3277]:
```suggestion
            assertTrue(archiveContains(archiveFolder.path(), "Depends on:", pr.id()));
            assertFalse(archiveContains(archiveFolder.path(), "Depends on:", depPr.id()));
```

**Review by edvbld — APPROVED** [review]
Looks good! Just a minor nit, I would have just used the wording `"Depends on"` instead of `"Depends on PR"` :smiling_face_with_three_hearts:

**Comment by rwestberg** [comment]
Sure, why not!

**Pull request closed without merging**